docs/ambiguity.md

```diff
@@ -1,79 +1,363 @@
-In `'later'` mode, the instant time will be the later instance of the duplicated time.
-In `'compatible'` mode, the same time is returned as `'earlier'` mode, which matches the behavior of existing JavaScript code that uses legacy `Date`.
+When date/time data for future times is stored with both the offset and the time zone, and if the time zone definition changes, then it's possible that the new time zone definition may conflict with previously-stored data.
+In this case, then the `offset` option to [`Temporal.LocalDateTime.from`](./localdatetime.md#from) is used to resolve the conflict:
 
+- `'use'`: Evaluate date/time values using the time zone offset if it's provided in the input.
+  This will keep the exact time unchanged even if local time will be different than what was originally stored.
+- `'ignore'`: Never use the time zone offset provided in the input. Instead, calculate the offset from the time zone.
+  This will keep local time unchanged but may result in a different exact time than was originally stored.
+- `'prefer'`: Evaluate date/time values using the offset if it's valid for this time zone.
+  If the offset is invalid, then calculate the offset from the time zone.
+  This option is rarely used when calling `from()`.
+  See the documentation of `with()` for more details about why this option is used.
+- `'reject'`: Throw a `RangeError` if the offset is not valid for the provided date and time in the provided time zone.
+
+The default is `reject` for [`Temporal.LocalDateTime.from`](./localdatetime.md#from) because there is no obvious default solution.
+Instead, the developer needs to decide how to fix the now-invalid data.
+
+For [`Temporal.LocalDateTime.with`](./localdatetime.md#with) the default is `'prefer'`.
+This default is helpful to prevent DST disambiguation from causing unexpected one-hour changes in exact time after making small changes to clock time fields.
+For example, if a [`Temporal.LocalDateTime`](./localdatetime.md) is set to the "second" 1:30AM on a day where the 1-2AM clock hour is repeated after a backwards DST transition, then calling `.with({minute: 45})` will result in an ambiguity which is resolved using the default `offset: 'prefer'` option.
+Because the existing offset is valid for the new time, it will be retained so the result will be the "second" 1:45AM.
+However, if the existing offset is not valid for the new result (e.g. `.with({hour: 0})`), then the default behavior will change the offset to match the new local time in that time zone.
+
+Note that offset vs. timezone conflicts only matter for [`Temporal.LocalDateTime`](./localdatetime.md) because no other Temporal type accepts both an IANA time zone and a time zone offset as an input to any method.
+For example, (`Temporal.Instant.from`)(./instant.md#from) will never run into conflicts because the [`Temporal.Instant`](./instant.md) type ignores the time sone in the input and only uses the offset.
```

Comment:
```suggestion
For example, [`Temporal.Instant.from`](./instant.md#from) will never run into conflicts because the [`Temporal.Instant`](./instant.md) type ignores the time sone in the input and only uses the offset.
```